src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,67 @@
+package codeu.controller;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Before;
+import org.mockito.Mockito;
+
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+
+import org.junit.Test;
+
+public class AdminServletTest {
+
+	private AdminServlet adminServlet;
+	private HttpServletRequest mockRequest;
+	private HttpServletResponse mockResponse;
+	private RequestDispatcher mockRequestDispatcher;
+	private ConversationStore mockConversationStore;
+	private MessageStore mockMessageStore;
+	private UserStore mockUserStore;
+
+	@Before
+	public void setup() {
+		adminServlet = new AdminServlet();
+		mockRequest = Mockito.mock(HttpServletRequest.class);
+		mockResponse = Mockito.mock(HttpServletResponse.class);
+		mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+		Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/adminpage.jsp")).thenReturn(mockRequestDispatcher);
+
+		mockConversationStore = Mockito.mock(ConversationStore.class);
+		adminServlet.setConversationStore(mockConversationStore);
+
+		mockMessageStore = Mockito.mock(MessageStore.class);
+		adminServlet.setMessageStore(mockMessageStore);
+
+		mockUserStore = Mockito.mock(UserStore.class);
+		adminServlet.setUserStore(mockUserStore);
+	}
+
+	@Test
+	public void testDoGet() throws IOException, ServletException {
+
+		int userCount = 0;
```

Comment:
Testing a 0-case is usually something you want to avoid, since default values for a lot of things are 0-like. Can you please change the tests to use more interesting information?

I think also, it would probably be more useful to test more of the code paths by adding users/messages/conversations to the various stores, and letting them return their values, rather than mock the `getNumUsers` methods, etc